### ISSUE TITLE
[P1] Excess initializer elements for arrays not diagnosed

### ISSUE THREAD

**Issue by ChaseWNorton** [anthropics/claudes-c-compiler#85]
## Summary

The compiler silently accepts array initializers with more elements than the array size. GCC/Clang emit `error: excess elements in array initializer`.

## Reproduction

```c
int arr[1] = {1, 2, 3};
```

```bash
ccc -c test.c -o /dev/null
# No warning or error — should diagnose excess elements
```

GCC output:
```
error: excess elements in array initializer
```

## Root cause

`src/frontend/sema/analysis.rs` — `analyze_initializer()` (~line 2236) recursively visits expressions in initializer lists but never validates that the number of initializers matches the declared array size.

## Expected behavior

The compiler should emit an error when an array initializer has more elements than the array can hold, matching GCC/Clang behavior.

## Acceptance criteria

- [ ] `int arr[1] = {1, 2, 3}` produces a diagnostic about excess elements
- [ ] `int arr[3] = {1, 2, 3}` still compiles cleanly (exact match)
- [ ] `int arr[3] = {1}` still compiles cleanly (fewer elements is valid — zero-fills rest)
- [ ] Test using `sema_error_count()` or `sema_warning_count()`

## Metadata

- **Reporter**: @lephe in #57
- **Phase**: Sema
- **Decomposed from**: #57

**Comment by ChaseWNorton** [comment]
**Triaged** — validated during decomposition of parent issue. Root cause identified, reproduction case provided, acceptance criteria defined. Ready for pickup.

**Comment by ChaseWNorton** [comment]
**Complete** — fix shipped in PR #120. Added excess initializer check in sema that compares initializer element count against declared array size.
